### DISCUSSION TITLE
why is `npx create-nx-workspace`even asking about my passphrase - explain or remove

### DISCUSSION THREAD

**Discussion by vicb** [nrwl/nx#27291] [Feature Requests]
When using `npx create-nx-workspace`:

![image](https://github.com/user-attachments/assets/ac448014-79f3-4d12-8c88-d067d0eb4c53)

I am being asked `Enter passphrase:`

Wait, what passphrase??

It could be confusing.  What about adding a prompt explaining what the CLI is trying to do and why exactly it needs the passphrase?

Thanks!

**Comment by mpsanchis** [comment]
Hi Victor!

Are you sure you're getting that message from Nx and not some other tool in your terminal? I have never seen such a prompt, and I can't find the word `passphrase` in the whole Nx codebase

**Reply by vicb** [reply]
I'm getting this error by running `npx create-nx-workspace` as indicated in my initial report.

I can think of several root cause:

- it tries to access a remote repo and the key has not been `ssh-add`ed
- I use a SSH key to [sign my commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#ssh-commit-signature-verification)

The second bullet point is most probably the root cause as I can't imagine `npx create-nx-workspace` accessing one of my remote repo.

> Are you sure you're getting that message from Nx and not some other tool in your terminal?

The message is definitely from `git` which is called by `npx create-nx-workspace` and that is exactly the problem here.
I was very surprised to see this message in the middle of the execution of `npx create-nx-workspace`.
So I guess it would be nice to add a message about who is sending this message (that is `git`) and what is `git` trying to do add that time. It should be prossible to do as this message should be emitted on the `stdout` of the process.

Message suggestion `While committing the code, git sent a message: "Enter passphrase:"`

Does this make sense?

**Reply by mpsanchis** [reply]
Sure, makes a lot of sense! I am upvoting this: if Nx is using git behind the scenes, it should handle its behaviour in different cases. 👍
